src/core/AutomationPattern.cpp

```diff
@@ -178,9 +178,19 @@ const AutomationPattern::objectVector& AutomationPattern::objects() const
 
 MidiTime AutomationPattern::timeMapLength() const
 {
-	if( m_timeMap.isEmpty() ) return 0;
+	MidiTime one_bar = MidiTime(1, 0);
+	if (m_timeMap.isEmpty())
+	{
+		return one_bar;
+	}
```

Comment:
I think this is a great candidate to be on one line:
```suggestion
	if (m_timeMap.isEmpty()) { return one_bar; }
```
Same for the other if. I would also drop the `one_bar` variable and directly return `MidiTime(1, 0)` in both cases. Reasoning:

- In some calls the value is never used.
- It's pretty reasonable to guess that MidiTime(1, 0) is one bar, or check if you really want to know. Even without knowing the specific length, it's clear that a placeholder value is returned for empty maps.
- Dropping the line keeps `timeMapLength` a little shorter. Alternatively, replacing it with a comment keeps the same length but could provide more insight. Hence you could argue the variable is the "worst of both worlds", haha.
  - `//Return one bar if length would otherwise be zero, to avoid invisible TCOs`

Reply:
Ah yeah, those can be one line!

As for `one_bar`, I was thinking more along DRY standard there.

Reply:
Yeah, I'm aware of DRY, I just don't think this is enough repetition to count. It's fine with me if you leave it though, this was my least important comment.